evcouplings/align/protocol.py

```diff
@@ -1003,6 +1002,145 @@ def jackhmmer_search(**kwargs):
 
     return outcfg
 
+def hmmbuild_and_search(**kwargs):
+    """
+    Protocol:
+
+    Hmmbuild and hmmsearch against a sequence database.
+    
+    Parameters
+    ----------
+    Mandatory kwargs arguments:
+        See list below in code where calling check_required
+
+    Returns
+    -------
+    outcfg : dict
+        Output configuration of the protocol, including
+        the following fields:
+
+        * target_sequence_file
+        * sequence_file
+        * raw_alignment_file
+        * hittable_file
+        * focus_mode
+        * focus_sequence
+        * segments
+    """
+    check_required(
+        kwargs,
+        [
+            "prefix", "sequence_id", "sequence_file",
+            "sequence_download_url", "region", "first_index",
+            "use_bitscores", "domain_threshold", "sequence_threshold",
+            "database", "iterations", "cpu", "nobias", "reuse_alignment",
+            "checkpoints_hmm", "checkpoints_ali", "hmmbuild",
+            "hmmsearch", "extract_annotation"
+        ]
+    )
+    prefix = kwargs["prefix"]
+
+    # make sure output directory exists
+    create_prefix_folders(prefix)
+
+    # store search sequence file here
+    target_sequence_file = prefix + ".fa"
+    full_sequence_file = prefix + "_full.fa"
+
+    # make sure search sequence is defined and load it
+    full_seq_file, (full_seq_id, full_seq) = fetch_sequence(
```

Comment:
Protocol shouldn't depend on a sequence given we are starting from an HMM (see separate discussion in pull request)